cmd/juju/space/space.go

```diff
@@ -235,3 +235,108 @@ func (c *SpaceCommandBase) RunWithSpaceAPI(ctx *cmd.Context, toRun RunOnSpaceAPI
 	defer api.Close()
 	return toRun(api, ctx)
 }
+
+// SubnetInfo is a source-agnostic representation of a subnet.
+// It may originate from state, or from a provider.
+type SubnetInfo struct {
```

Comment:
Now that we have mirrored this type for display, is it worth adding an equality test between the core type and the display type that will break with a "you didn't consider this" message if they diverge?

Reply:
Well, the idea is that they're locked stepped at the moment, but that shouldn't always be the case. This what this PR is addressing. They just happen to be the same at the moment.